src/errors/UndefinedProtectedPrivateKeyError.ts

```diff
@@ -0,0 +1,8 @@
+class UndefinedProtectedPrivateKeyError extends Error {
+  constructor(msg: string) {
+    super(msg);
+    Object.setPrototypeOf(this, UndefinedProtectedPrivateKeyError.prototype);
```

Comment:
It is a question, why does it call [`Object.setPrototypeOf`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/setPrototypeOf)? (P.S. I read [Mozilla Custom Error Types section](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error#custom_error_types).)

Reply:
I'm curious too.🤔

Reply:
I agree.